trpcultivate_phenotypes/src/Service/TripalCultivatePhenotypesTraitsService.php

```diff
@@ -255,163 +308,286 @@ public function insertTrait($trait, $schema = NULL) {
         'unit'  => $arr_trait['unit']['id']
       ];
     }
-    else {
-      return 0;
-    }
   }
 
   /**
    * Get trait.
    *
-   * @param array $trait
-   *   Key:
-   *     id - get trait by id number (cvterm_id) or
-   *     name - get trait by name (cvterm.name).
+   * @param string|int $trait
+   *   A string value is the trait name (cvterm.name), whereas an integer value
+   *   is the trait id number (cvterm.cvterm_id).
    *
    * @return object
-   *   Matching record/line in cvterm table.
+   *   Matching record/line in cvterm table or 0 if trait record Was not found.
+   *
+   * @dependencies
+   *   getTraitAsset()
    */
   public function getTrait($trait) {
-    // Configuration settings of the genus.
-    $genus_config = $this->config;
+    $trait_rec = $this->getTraitAsset($trait, 'trait');
```

Comment:
Just adding a comment here to make it more clear why this function literally just calls another. 🤪 
```suggestion
    // Since the trait is simply a cvterm, we can use our helper method to retrieve it.
    $trait_rec = $this->getTraitAsset($trait, 'trait');
```